apps/web/src/hooks/useBestAMMTrade.ts

```diff
@@ -801,6 +948,38 @@ type PrefetchParams = {
 function getCurrencyIdentifierForApi(currency: Currency) {
   return currency.isNative ? zeroAddress : currency.address
 }
+export function useTradeApiPrefetchShadow(
+  { currencyA, currencyB, poolTypes, enabled = true }: PrefetchParams,
+  prefix: string,
+  queryType: string,
+) {
+  return useQuery({
+    enabled: !!(currencyA && currencyB && poolTypes?.length && enabled),
+    queryKey: [`${queryType}-prefetch`, currencyA?.chainId, currencyA?.symbol, currencyB?.symbol, poolTypes] as const,
+    queryFn: async ({ signal }) => {
+      if (!currencyA || !currencyB || !poolTypes?.length) {
+        throw new Error('Invalid prefetch params')
+      }
+
+      const serverRes = await fetch(
+        `${prefix}/_pools/${currencyA.chainId}/${getCurrencyIdentifierForApi(currencyA)}/${getCurrencyIdentifierForApi(
+          currencyB,
+        )}?${qs.stringify({ protocols: poolTypes.map(getPoolTypeKey) })}`,
+        {
+          method: 'GET',
+          signal,
+        },
+      )
+      const res = await serverRes.json()
+      if (!res.success) {
+        throw new Error(res.message)
+      }
+      return res
+    },
+    staleTime: currencyA?.chainId ? POOLS_FAST_REVALIDATE[currencyA.chainId] : 0,
+    refetchInterval: currencyA?.chainId ? POOLS_FAST_REVALIDATE[currencyA.chainId] : 0,
+  })
+}
```

Comment:
we don't need a new hook. Let's reuse `useTradeApiPrefetch`

Reply:
cuz it's not used right now

Reply:
done